src/fundamental/macro-fundamental.h

```diff
@@ -345,6 +345,20 @@ static inline size_t ALIGN_TO(size_t l, size_t ali) {
                 ((l) + (ali) - 1) & ~((ali) - 1),                      \
                 VOID_0)
 
+/* Similar to ((t *) (void *) (p)) to cast a pointer. The macro
+ * asserts that the pointer has a suitable alignment for type "t".
+ * This exists for places where otherwise "-Wcast-align=strict"
+ * would issue a warning or if you want to assert that the cast
+ * gives a pointer of suitable alignment. */
+#define PTRFORCEALIGN(t, p)                                             \
+        ({                                                              \
+                const void * const _p = (p);                            \
```

Comment:
hmm, the second "const" doesn't really get us anything, does it? it's not tightening any checks of stuff passed into this macro, or am I missing something?